targets/unity-v2/source/Shared/Internal/Util.cs

```diff
@@ -74,16 +74,66 @@ public static string Format(string text, params object[] args)
         /// </summary>
         public static string ReadAllFileText(string filename)
         {
+            if (!File.Exists(filename))
+                return string.Empty;
+                
             if (_sb == null)
                 _sb = new StringBuilder();
             _sb.Length = 0;
 
-            var fs = new FileStream(filename, FileMode.Open);
-            var br = new BinaryReader(fs);
-            while (br.BaseStream.Position != br.BaseStream.Length)
-                _sb.Append(br.ReadChar());
+            FileStream fs = null;
+            try 
+            {
+                fs = new FileStream(filename, FileMode.Open);
+                using (BinaryReader br = new BinaryReader(fs))
+                {
+                    // Attempt to release the descriptor immediately
+                    fs = null;
 
+                    while (br.BaseStream.Position != br.BaseStream.Length)
+                        _sb.Append(br.ReadChar());
+                }
+            }
+            finally
+            {
+                // Make sure to dispose of the file stream in case the construction of binary reader fails
+                if (fs != null)
+                    fs.Dispose();
+            }
+            
             return _sb.ToString();
         }
+
+        internal static string GetLocalSettingsFileProperty(string propertyKey)
+        {
+            string envFileContent = null;
+            
+            string envFileName = "playfab.local.settings.json";
```

Comment:
Should we move this out to be a private field like in the C# SDK?

Reply:
Done